vyperlang/titanoboa
# Implement RPC calls in Jupyter/Colab

Fixes #119
Depends on https://github.com/vyperlang/titanoboa/pull/132

### What I did
Extended the interface with the browser to support RPC calls.

### How I did it
It includes two new classes on the python side:
- `BrowserRpc` calls the browser when a RPC call is made
- `boa.set_browser()` combines `BrowserRpc` and `BrowserSigner` together to allow the environment to be used in boa
- `boa.set_network(url)` works as an alternative to `boa.set_env(NetworkEnv(url))`

The JavaScript uses [this method](https://github.com/ethers-io/ethers.js/blob/a26ff77c210660e641455150ccea6c0530f02f61/lib.esm/providers/provider-browser.d.ts#L47) from ethers.js.

Tests are included, but unfortunately I had to mock all jupyter and tornado dependencies - they aren't installed in boa normally.

### How to verify it
Install this branch and then run in Jupyter or Google Colab:
```
import boa
boa.set_browser()
```
- `boa.env.eoa` should return your wallet address.
- `boa.env.get_gas_price()` returns the current gas price.
- Deploying and calling contracts should still work

![image](https://github.com/vyperlang/titanoboa/assets/2369243/581b19fb-48a9-471f-b1a6-53006fda5929)

### Description for the changelog
- Add new `boa.set_browser()` interface to interact with the browser wallet RPC in JupyterLab

### Cute Animal Picture
![image](https://github.com/vyperlang/titanoboa/assets/2369243/d9c80311-9510-43ef-b36b-cce2713bfe7a)
